src/plotrecipes.jl

```diff
@@ -1,60 +1,64 @@
 using DimensionalData: refdims_title
 
+const GeoDim = Union{GeoXDim,GeoYDim,GeoZDim}
+
 struct GeoPlot end
 
-# We only look at arrays with Lat/Lon here.
+# We only look at arrays with <:GeoXDim/<:GeoYDim here.
 # Otherwise they fall back to DimensionalData.jl recipes
 @recipe function f(A::AbstractGeoArray)
     A = GeoArray(A)
-    if all(hasdim(A, (Lat(), Lon())))
+    if all(hasdim(A, (GeoXDim, GeoYDim))) || all(hasdim(A, (GeoXDim, GeoZDim))) || all(hasdim(A, (GeoYDim, GeoZDim))) || hasdim(A, GeoZDim)
         # Heatmap or multiple heatmaps. Use GD recipes.
         GeoPlot(), prepare(A)
     else
-        # This is not a Lat/Lon heatmap. Fall back to DD recipes after reprojecting
+        # This is not a GeoXDim/GeoYDim heatmap. Fall back to DD recipes after reprojecting
         da = A |> GeoArray |> a -> DimArray(a; dims=_maybe_mapped(dims(a)))
         DimensionalData.DimensionalPlot(), da
     end
 end
 
 # Plot 3d arrays as multiple tiled plots
-@recipe function f(::GeoPlot, A::GeoArray{T,3,<:Tuple{<:Lat,<:Lon,D}}) where {T,D}
+@recipe function f(::GeoPlot, A::GeoArray{T,3,<:Tuple{<:GeoDim,<:GeoDim,D}}) where {T,D}
     nplots = size(A, 3)
     if nplots > 1
         :layout --> nplots
-        :xguide --> permutedims(string.(val(dims(A, D))))
+        :title --> permutedims(string.(val(dims(A, D))))
         for i in 1:nplots
             @series begin
                 seriestype := :heatmap
                 aspect_ratio := 1
                 subplot := i
                 slice = A[:, :, i]
-                lats, lons = map(prepare, dims(slice))
-                lons, lats, parent(slice)
+                xs, ys = map(prepare, dims(slice))
+                xs, ys, permutedims(slice) |> parent
             end
         end
     else
         GeoPlot(), A[:, :, 1]
     end
 end
 
-# # Permute for correct Lat/Lon order
-@recipe function f(::GeoPlot, A::GeoArray{T,3}) where {T}
-    GeoPlot(), permutedims(A, (Lat, Lon, Dimension))
-end
-
 # # Plot a sinlge 2d map
-@recipe function f(::GeoPlot, A::GeoArray{T,2,<:Tuple{<:Lat,<:Lon}}) where T
+@recipe function f(::GeoPlot, A::GeoArray{T,2,<:Tuple{<:GeoDim,<:GeoDim}}) where T
     :seriestype --> :heatmap
     :aspect_ratio --> 1
     :colorbar_title --> name(A)
     :title --> refdims_title(A)
-    lats, lons = map(prepare, dims(A))
-    lons, lats, parent(A)
+    :xguide --> name(dims(A, 1))
+    :yguide --> name(dims(A, 2))
+    x1, x2 = map(prepare, dims(A))
+    x1, x2, permutedims(A) |> parent
 end
 
-# # Permute for correct Lat/Lon order
-@recipe function f(::GeoPlot, A::GeoArray{T,2,<:Tuple{<:Lon,<:Lat}}) where T
-    GeoPlot(), permutedims(A)
+# # Plot a vertical 1d line
+@recipe function f(::GeoPlot, A::GeoArray{T,1,<:Tuple{<:GeoZDim}}) where T
```

Comment:
Does this need to be in GeoData? should all DimensionalData `Z` plot vertically? (I have no idea)

Reply:
Ah I'm not sure but at least in oceanography and atmospheric science it's very common to want to plot a vertical profile (vector or 1D array) of some quantity, in which case it's nice to have the vertical coordinate be on the y-axis so the profile is oriented in an intuitive way (see the vertical profile plot I posted above).

By default the z-coordinate will go from left to right which makes interpreting vertical profiles a little tougher (unless you tilt your head sideways 🙃).